### PR TITLE
Vector tests

### PR DESCRIPTION
**Pull request by Lobotuerk**
Vector tests finished, nits in the original Isometry test, and last lint changes on the original isometry lib

### CODE REVIEW

File: course/test/src/vector3_TEST.cc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#include <isometry/isometry.hpp>
+
+namespace ekumen {
+namespace math {
+namespace test {
+namespace {
+
+  GTEST_TEST(Vector3Test, Vector3FullTests) {
+    const double kTolerance{1e-12};
+    const Vector3 p{1., 2., 3.};
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc:23]: Are we testing the default constructor `Vector3()`  anywhere?

File: course/test/src/vector3_TEST.cc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    EXPECT_EQ(p + q, std::initializer_list<double>({5., 7., 9.}));
+    EXPECT_EQ(p - q, std::initializer_list<double>({-3., -3., -3.}));
+    EXPECT_EQ(p * q, std::initializer_list<double>({4., 10., 18.}));
+    EXPECT_EQ(p / q, std::initializer_list<double>({.25, .4, .5}));
+    EXPECT_EQ(p * 2., Vector3(2., 4., 6.));
+    EXPECT_EQ(q / 2., Vector3(2., 2.5, 3.));
+    EXPECT_EQ(2 * q, Vector3(8., 10., 12.));
+
+    r += q;
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc]: Please add tests for the return value of these with-assignment operators.

File: course/test/src/vector3_TEST.cc [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    r /= 2;
+    EXPECT_EQ(r, std::initializer_list<double>({1., 1., 1.}));
+
+    r = Vector3(1., 2., 3.);
+    EXPECT_TRUE(r == p);
+    EXPECT_TRUE(r != q);
+
+    EXPECT_NEAR(p.norm(), 3.7416573867739413, kTolerance);
+    EXPECT_EQ(p.x(), 1.);
+    EXPECT_EQ(p.y(), 2.);
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc:73]: We need to test also the non-const versions of these getters.

File: course/test/src/vector3_TEST.cc [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    r /= q;
+    EXPECT_EQ(r, std::initializer_list<double>({1., 1., 1.}));
+    r *= 2;
+    EXPECT_EQ(r, std::initializer_list<double>({2., 2., 2.}));
+    r /= 2;
+    EXPECT_EQ(r, std::initializer_list<double>({1., 1., 1.}));
+
+    r = Vector3(1., 2., 3.);
+    EXPECT_TRUE(r == p);
+    EXPECT_TRUE(r != q);
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc:67]: I think we are not testing the `operator!=(const std::initializer _list &)` operator.

File: course/test/src/vector3_TEST.cc [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    EXPECT_NEAR(p.norm(), 3.7416573867739413, kTolerance);
+    EXPECT_EQ(p.x(), 1.);
+    EXPECT_EQ(p.y(), 2.);
+    EXPECT_EQ(p.z(), 3.);
+    EXPECT_EQ(p[0], 1.);
+    EXPECT_EQ(p[1], 2.);
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc:76]: We need to test also the non-const versions of this operator.

File: course/test/src/vector3_TEST.cc [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    EXPECT_EQ(p[2], 3.);
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc:77]: Please add tests for the out-of-range case.

File: course/test/src/vector3_TEST.cc [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    std::stringstream ss;
+    ss << p;
+    EXPECT_EQ(ss.str(), "(x: 1, y: 2, z: 3)");
+
+    Vector3 t;
+    EXPECT_EQ(t, Vector3::kZero);
+    t.x() = 1.;
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc:88]: We need to test all three of these setters.

File: course/test/src/vector3_TEST.cc [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    t[1] = 2.;
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc]: Test also the out-of-range case.

File: course/test/src/vector3_TEST.cc [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    EXPECT_TRUE(Vector3::kUnitZ == std::initializer_list<double>({0., 0., 1.}));
+    EXPECT_TRUE(Vector3::kUnitX != std::initializer_list<double>({1., 1., 0.}));
+    EXPECT_TRUE(Vector3::kUnitY != std::initializer_list<double>({0., 1., 1.}));
+    EXPECT_TRUE(Vector3::kUnitZ != std::initializer_list<double>({1., 0., 1.}));
+    EXPECT_TRUE(Vector3::kUnitZ == Vector3::kUnitX.cross(Vector3::kUnitY));
+    EXPECT_TRUE(Vector3::kUnitY == Vector3::kUnitZ.cross(Vector3::kUnitX));
+    EXPECT_TRUE(Vector3::kUnitX == Vector3::kUnitY.cross(Vector3::kUnitZ));
+    EXPECT_NEAR(Vector3::kUnitX.dot(Vector3::kUnitZ), 0., kTolerance);
+    EXPECT_NEAR(Vector3::kUnitX.dot(Vector3::kUnitY), 0., kTolerance);
+    EXPECT_NEAR(Vector3::kUnitY.dot(Vector3::kUnitZ), 0., kTolerance);
```
Comment (glpuga) [inline course/test/src/vector3_TEST.cc:41]: If `Vector3::dot()` were just a `return 0.0;` we would pass all these tests. Please add tests for cases that result in non-zero values.

**Review by glpuga — CHANGES_REQUESTED** [review]
Good work!

File: course/test/src/vector3_TEST.cc (hunk 1 above)
Comment (Lobotuerk) [reply course/test/src/vector3_TEST.cc:23]: Line 81, is that not testing the default constructor?

**Review by glpuga — APPROVED** [review]
Great work! LGTM

**Comment by glpuga** [comment]
Approved, please squash and merge!
